pulser-core/pulser/sequence/_schedule.py

```diff
@@ -78,26 +77,38 @@ def adjust_duration(self, duration: int) -> int:
                 max(duration, self.channel_obj.min_duration)
             )
 
-    def get_samples(self, N: int) -> samples.ChannelSamples:
+    def get_samples(self, modulated: bool = False) -> ChannelSamples:
         """Returns the samples of the channel.
 
         Args:
-            N (int): the duration of the parent sequence. Necessary to give the
-            right length to the sample arrays.
+            modulated: Whether to return the modulated samples.
         """
         # Keep only pulse slots
+        dt = self.get_duration()
         channel_slots = [s for s in self.slots if isinstance(s.type, Pulse)]
```

Comment:
Swapping lines for consistent comment

```suggestion
        channel_slots = [s for s in self.slots if isinstance(s.type, Pulse)]
        dt = self.get_duration()
```